wkcuber/converter.py

```diff
@@ -0,0 +1,283 @@
+from argparse import ArgumentParser, Namespace
+from os import path, cpu_count, sep
+from pathlib import Path
+from natsort import natsorted
+from .utils import find_files, add_scale_flag
+from typing import Iterable, List, Any, Tuple, cast, Dict
+from .convert_nifti import main as convert_nifti
+from .convert_knossos import main as convert_knossos
+from .__main__ import main as convert_image_stack
+
+from .image_readers import image_reader
+
+
+def create_parser() -> ArgumentParser:
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "source_path", help="Input file or directory containing the input files."
+    )
+
+    parser.add_argument(
+        "target_path", help="Output directory for the generated dataset."
+    )
+
+    add_scale_flag(parser)
+
+    return parser
+
+
+def put_default_if_not_present(args: Namespace, name: str, default: Any) -> None:
+    dictionary = vars(args)
+    if name not in dictionary:
+        dictionary[name] = default
+
+
+def add_or_update_if_present(dictionary: dict, name: str, value: Any) -> None:
+    if name not in dictionary:
+        dictionary[name] = {value}
+    else:
+        dictionary[name].add(value)
+
+
+def get_source_files(
+    input_path: str, extensions: Iterable[str], allows_single_file_input: bool
+) -> List[str]:
+    source_path = Path(input_path)
+
+    if source_path.is_dir():
+        input_path = path.join(input_path, "**")
+    elif not allows_single_file_input:
+        return []
+
+    source_files = list(find_files(input_path, extensions))
+
+    return source_files
+
+
+class Converter:
+    def accepts_input(self, source_path: str) -> bool:
+        pass
+
+    def convert_input(self, args: Namespace) -> None:
+        pass
+
+
+class WkwConverter(Converter):
+    def accepts_input(self, source_path: str) -> bool:
+        source_files = get_source_files(source_path, {".wkw"}, False)
+        return len(source_files) > 0
+
+    def convert_input(self, args: Namespace) -> None:
+        print("Already a WKW dataset. No conversion necessary...")
```

Comment:
This is a more general question for me: Do we want to have so much console output? I created these `print` calls as validation for me, but maybe they are interesting for the user as well.

Reply:
I think, high-level logging which shows which "insights" the cuber has gathered and which actions it intends to make is useful. I think one needs to decide case-by-case if something is useful. Usually, it's best to validate this by running the cuber and inspecting whether the output seems useful (instead of going through the logging code).

If something is too detailed, one can also switch from `logger.info` to `logger.debug`. Then, that information is only printed when `--verbose` is used :)